library/alloc/src/raw_vec/mod.rs

```diff
@@ -409,8 +409,9 @@ unsafe impl<#[may_dangle] T, A: Allocator> Drop for RawVec<T, A> {
 
 impl<A: Allocator> RawVecInner<A> {
     #[inline]
-    const fn new_in(alloc: A, align: usize) -> Self {
-        let ptr = unsafe { core::mem::transmute(align) };
+    const fn new_in(alloc: A, align: Alignment) -> Self {
+        // SAFETY: `Alignment` is non-zero.
+        let ptr = unsafe { Unique::new_unchecked(ptr::without_provenance_mut(align.as_usize())) };
```

Comment:
`new_unchecked` inserts UB checks that can hinder optimization. Using these safe helpers will avoid that:
```suggestion
        let ptr = Unique::from(NonNull::without_provenance(align.as_nonzero()));
```

Reply:
Perfect, thanks - this fn is `const` so I can't use `Unique::from`, are you okay with a transmute?

Reply:
Sure! Or you could add a function like `from_non_null` to `Unique`, make that `const` and use it here and in the `From` `impl`.